plenum/server/replica_freshness_checker.py

```diff
@@ -0,0 +1,58 @@
+from collections import OrderedDict
+
+
+class FreshnessState():
+
+    def __init__(self, last_updated, last_marked_as_outdated) -> None:
+        self.last_updated = last_updated
+        self.last_marked_as_outdated = last_marked_as_outdated
+
+
+class FreshnessChecker():
+
+    def __init__(self,
+                 freshness_timeout):
+        self.freshness_timeout = freshness_timeout
+        self._ledger_freshness = {}  # Dict[ledger_id -> FreshnessState]
+        self._outdated_ledgers = OrderedDict()
+
+    def register_ledger(self, ledger_id, initial_time):
+        self._ledger_freshness[ledger_id] = FreshnessState(initial_time, initial_time)
+
+    def check_freshness(self, ts):
+        '''
+        Get all ledger IDs for which
+          A) not updated for more than Freshness Timeout
+          B) hasn't been checked for more than than Freshness Timeout
+
+        :param ts: the current time check the freshness against
+        :return: None
```

Comment:
According to description of this function (get all ledger IDs...) returning None seems a bit strange. After looking at usage pattern it seems like `self._outdated_ledgers` and related API can be removed while `check_freshness` changed to actually return list of outdated ledgers. This will remove unnecessary state from class and simplify API a lot in my opinion.